integration/helm_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/integration/skaffold"
+	"github.com/google/go-cmp/cmp"
+)
+
+const (
+	TestVersion = "vtest"
+)
+
+func TestHelmDeploy(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	if os.Getenv("REMOTE_INTEGRATION") != "true" {
+		t.Skip("skipping remote only test")
+	}
+
+	helmDir := "examples/helm-deployment"
+
+	ns, client, deleteNs := SetupNamespace(t)
+	// To fix #1823, we make use of env variable templating for release name
+	env := []string{fmt.Sprintf("TEST_NS=%s", ns.Name)}
+	depName := fmt.Sprintf("skaffold-helm-%s", ns.Name)
+	fmt.Println(env)
+	defer func() {
+		skaffold.Delete().InDir(helmDir).InNs(ns.Name).WithEnv(env).RunOrFail(t)
+		deleteNs()
+	}()
+
+	skaffold.Deploy("--images", "gcr.io/k8s-skaffold/skaffold-helm").InDir(helmDir).InNs(ns.Name).WithEnv(env).RunOrFailOutput(t)
+
+	client.WaitForDeploymentsToStabilize(depName)
+
+	expectedLabels := map[string]string{
+		"app.kubernetes.io/managed-by": TestVersion,
+		"release":                      depName,
+		"skaffold.dev/deployer":        "helm",
+	}
+
+	// check if labels are set correctly for deploument
+	dep := client.GetDeployment(depName)
+	if d := diffLabels(expectedLabels, dep.ObjectMeta.Labels); d != "" {
+		t.Errorf("did not find expected labels for dep %s: %s", depName, d)
+	}
+}
+
+func diffLabels(expected map[string]string, actual map[string]string) string {
+	extracted := map[string]string{}
+	for k, v := range actual {
+		switch k {
+		case "app.kubernetes.io/managed-by":
```

Comment:
this is on purpose. Using constants in test code,
- reduces readability
- do not want to re-use the constant already defined in code because then you are not testing for the correctness of the string.